### PR TITLE
ISSUE #328

### PR DESCRIPTION
**Pull request by jancris100**
✅ Closes https://github.com/undefinedorgcr/gostarkme-backend/issues/328
✅ Added unit tests for legacy map
✅ Update Test Cases for update_receive_donation Method
✅ Added boundary and Edge Cases:

Implemented Test Cases:

✅ Legacy Map Tests:
Verify that a new donator is correctly registered with their initial donation amount.
Verify that an existing donator's total donation is updated upon subsequent donations.

✅ update_receive_donation Method:
Ensure the method updates the legacy map before emitting the donation event.
Validate that the donation event is emitted with correct parameters.

✅ Boundary & Edge Cases:
Tested a new donator contributing a small amount.
Tested an existing donator contributing a large amount.
Verified behavior when multiple donations are made in rapid succession.

image

### CODE REVIEW

**Comment by jancris100** [comment]
Ya no hice cambios en el fund.cairo porque ahora podia usar la funcion que agregaron get_single_donator_by_address que basicamente cumplia lo que yo habia realizado, y funcione los metodos en los test porque eran muchos y a veces tiraba error de  JSON-RPC error por cantidad de consultas. 

The new tests are test_donation_scenarios, that have 3 methods  
• Test with a new donator contributing a small amount.

• Test with an existing donator contributing a large amount.

• Verify behavior when multiple donations are made in rapid succession 
and 
test_donator_registration_and_subsequent_donations that have 2 methods 

• Verify that a new donator is correctly registered in the legacy map with their initial donation amount.

• Verify that an existing donator’s total donation amount is updated correctly upon subsequent donations.
And for the last fn was updated test_update_received_donation with the requirementes

**Review by EmmanuelAR — CHANGES_REQUESTED** [review]

File: tests/test_fund.cairo [hunk 1]
```diff
@@ -3,6 +3,7 @@
 // ***************************************************************************************
 use starknet::{ContractAddress, contract_address_const};
 use starknet::syscalls::call_contract_syscall;
+use starknet::testing::set_block_timestamp;
```
[the remark this answers is not in the document]
Comment (jancris100) [reply tests/test_fund.cairo]: Ready i deleted

[the remark this answers is not in the document]
Comment (jancris100) [reply tests/test_fund.cairo]: @EmmanuelAR

**Review by EmmanuelAR — APPROVED** [review]
lgtm great work Sr
